File: AtlasWeb/Core/Modules/Tibbi_Surec/Tetkik_Istem_Modulu/AdditionalReportFormViewModel.ts
//$CC532205
import { BaseViewModel } from "NebulaClient/Model/BaseViewModel";
import { AdditionalReport } from "NebulaClient/Model/AtlasClientModel";
import { Type } from 'NebulaClient/ClassTransformer';

export class AdditionalReportFormViewModel extends BaseViewModel {
     protected __type__: string = 'Core.Models.AdditionalReportFormViewModel, Core';
    @Type(() => AdditionalReport)
    public _AdditionalReport: AdditionalReport = new AdditionalReport();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Tetkik_Istem_Modulu/AdditionalReportFormViewModel.cs
//$63EDC916
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using Core.Controllers;

namespace Core.Controllers
{
    public partial class AdditionalReportServiceController
    {
    }
}

namespace Core.Models
{
    public partial class AdditionalReportFormViewModel: BaseAdditionalInfoFormViewModel
    {
        public override BaseAdditionalInfo AddViewModelToContext(TTObjectContext objectContext)
        {
            var additionalReportFormService = new AdditionalReportServiceController();
            additionalReportFormService.AdditionalReportFormInternal(this, objectContext);
            return (AdditionalReport)objectContext.GetObject(this._AdditionalReport.ObjectID, "AdditionalReport");
        }
    }
}
